libraries/iio/iio.c

```diff
@@ -698,6 +733,21 @@ static ssize_t iio_transfer_dev_to_mem(const char *device, size_t bytes_count)
 		return iio_interface->dev_descriptor->transfer_dev_to_mem(
 			       iio_interface->dev_instance,
 			       bytes_count, iio_interface->ch_mask);
+	//else
+	struct iio_data_buffer	*r_buff;
+	uint32_t		samples;
+	ssize_t			ret;
+
+	r_buff = iio_interface->read_buffer;
+	if (r_buff && iio_interface->dev_descriptor->read_dev) {
+		if (bytes_count > r_buff->size)
+			return -ENOMEM;
+		samples = bytes_to_samples(iio_interface, bytes_count);
+		ret = iio_interface->dev_descriptor->read_dev(
```

Comment:
This function looks very similar to transfer_dev_to_mem

Reply:
Maybe we should try and have a single function pointer with diffrent implementations

Reply:
OK, they are removed :)